### DISCUSSION TITLE
Get list of files from a specific branch and compare it with main branch using curl

### DISCUSSION THREAD

**Discussion by azzarb** [community/community#111325] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

General

### Body

Hi, 
Is there a way, where we can check/list all the files from a specific branch (other than master/main/default) using curl api and I want to parse the data using jq..I can get all the files which are inside ".github/workflows" but i want to revert to that specific branch and get all the files in it.
Thanks

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @azzarb ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. I've gone ahead and moved it for you. Good luck!
